include/ginkgo/core/base/math.hpp

```diff
@@ -362,7 +362,7 @@ GKO_INLINE GKO_ATTRIBUTES constexpr int64 ceildiv(int64 num, int64 den)
 template <typename T>
 GKO_INLINE __host__ constexpr T zero()
 {
-    return T(0);
+    return T{0};
```

Comment:
nit: To be consistent with the `__device__` function:
```suggestion
    return T{};
```